lms/static/scripts/postmessage_json_rpc/test/server-test.js

```diff
@@ -124,6 +124,70 @@ describe('Server', () => {
     });
   });
 
+  describe('when a valid notification request is received', () => {
+    beforeEach(() => {
+      const errorMethod = sinon.stub().throws();
+      server.register('errorMethod', errorMethod);
+      sinon.stub(console, 'error');
+    });
+
+    afterEach(() => {
+      console.error.restore();
+    });
+
+    it('calls the registered method', async () => {
+      window.postMessage(validNotificationRequest(), serversOrigin);
+      await new Promise(resolve => setTimeout(resolve, 0));
```

Comment:
There is a `delay` function in `test-util/wait.js` that could be used here to shorten the code a little. Search for existing uses of `await delay(0)` in the codebase.